mindsdb/integrations/handlers/hsqldb_handler/hsqldb_handler.py

```diff
@@ -0,0 +1,232 @@
+        Returns:
+            HandlerStatusResponse
+        """
+
+        if self.is_connected is True:
+            return self.connection
+
+        self.connection = pyodbc.connect(self.conn_str, timeout=10)
+        self.is_connected = True
+
+        return self.connection
+
+    def disconnect(self):
+        """
+        Close any existing connections.
+        """
+
+        if self.is_connected is False:
+            return
+
+        self.connection.close()
+        self.is_connected = False
+        return self.is_connected
+
+    def check_connection(self) -> StatusResponse:
+        """
+        Check connection to the handler.
+        Returns:
+            HandlerStatusResponse
+        """
+
+        response = StatusResponse(False)
+        need_to_close = self.is_connected is False
+
+        try:
+            self.connect()
+            response.success = True
+        except Exception as e:
+            log.logger.error(f'Error connecting to SQLite, {e}!')
+            response.error_message = str(e)
+        finally:
+            if response.success is True and need_to_close:
+                self.disconnect()
+            if response.success is False and self.is_connected is True:
+                self.is_connected = False
+
+        return response
+
+    def native_query(self, query: str) -> StatusResponse:
+        """
+        Receive raw query and act upon it somehow.
+        Args:
+            query (str): query in native format
+        Returns:
+            HandlerResponse
+        """
+
+        need_to_close = self.is_connected is False
+
+        connection = self.connect()
+        with connection.cursor() as cursor:
+            try:
+                cursor.execute(query)
+                result = cursor.fetchall()
+                if result:
+                    response = Response(
+                        RESPONSE_TYPE.TABLE,
+                        data_frame=pd.DataFrame.from_records(
+                            result,
+                            columns=[x[0] for x in cursor.description]
+                        )
+                    )
+
+                else:
+                    response = Response(RESPONSE_TYPE.OK)
+                    connection.commit()
+            except Exception as e:
+                log.logger.error(f'Error running query: {query}!')
+                response = Response(
+                    RESPONSE_TYPE.ERROR,
+                    error_message=str(e)
+                )
+
+        if need_to_close is True:
+            self.disconnect()
+
+        return response
+
+    def query(self, query: ASTNode) -> StatusResponse:
+        """
+        Receive query as AST (abstract syntax tree) and act upon it somehow.
+        Args:
+            query (ASTNode): sql query represented as AST. May be any kind
+                of query: SELECT, INTSERT, DELETE, etc
+        Returns:
+            HandlerResponse
+        """
+
+        renderer = SqlalchemyRender('postgres')

```

Comment:
Is hsqldb postgresql compatible?

Reply:
Hello, yes it is compatible as stated in the official page, you can read more about it here in the [compatibility page](https://hsqldb.org/doc/2.0/guide/compatibility-chapt.html) as well as in the [HyperSQL ODBC documentation](http://hsqldb.org/doc/2.0/guide/guide.html#odbc-chapt) in the "Overview" section of the page

I used HSQL database manager to create a database and connect it to MindsDB. The default demo database works too with this handler since it uses the Postgres driver.

<img width="777" alt="image" src="https://user-images.githubusercontent.com/87588133/224382311-e5e04a3e-cc84-40ef-85c5-3461d10d14ee.png">
![image](https://user-images.githubusercontent.com/87588133/224389909-526006a1-1b14-40bb-94ec-156cda05ed6c.png)